pkg/volume/volume.go

```diff
@@ -177,8 +177,8 @@ type Attacher interface {
 
 	// VolumesAreAttached checks whether the list of volumes still attached to the specified
 	// the node. It returns a map which maps from the volume spec to the checking result.
-	// If an error is occured during checking, the error will be returned
-	VolumesAreAttached(specs []*Spec, nodeName types.NodeName) (map[*Spec]bool, error)
+	// If an error is occurred during checking, the error will be returned
+	VolumesAreAttached(volumesByNode map[types.NodeName][]*Spec) (map[*Spec]bool, error)
```

Comment:
I think @jsafrane is right. We have to change the returned data structure. For example, we have to nodes and both attach volume 1. If volume is detached from node 1, but still attach to node 2, VolumesAreAttached might return attached to either true or false, which neither of them is correct. 
I think we have to use map[types.NodeName][map[*Spec]bool]?